src/devices/cpu/powerpc/ppccom.cpp

```diff
@@ -752,6 +755,8 @@ void ppc_device::device_start()
 
 	m_tb_divisor = (m_tb_divisor * clock() + m_system_clock / 2 - 1) / m_system_clock;
 
+	m_serial_clock = std::min(c_serial_clock != 0 ? c_serial_clock : 3686400, m_system_clock / 2); // serial clock can be at most half the system clock
+
```

Comment:
Would it be better to throw a fatal error or something if the serial clock frequency is too high?  In reality, if you feed it a frequency too high you’ll get aliasing from the synchronisation circuit so your clock intervals will be uneven – you won’t get a neat half system clock rate.

Reply:
I considered a fatal error here initially but I wasn't sure if it'd be ok. I'll change it. It'd be easier and safer to catch and fix bad clocks during testing.

Reply:
Pushed the change.

Tested it with a large clock and acceptable clock and it's working as expected, but throwing a fatal error in `device_start` throws a stack trace in addition to the fatal error message fyi. I tested another device's `device_start` and it does the same thing there, and tested in another place that wasn't a `device_start` and it errors without the stack trace. I don't know if this is expected behavior or not but just wanted to let you know.

```
Fatal error: PPC: serial clock (117964800) must not be more than half of the system clock (32000000)

-----------------------------------------------------
Exception at EIP=00007ff788b741c0 (not found): ACCESS VIOLATION
While attempting to read memory at 0000000000000090
-----------------------------------------------------
RAX=00007ff78bf47f50 RBX=0000000000000000 RCX=0000000000000000 RDX=000001eed8c10000
RSI=000001eedb2a5b30 RDI=000001eedd7742f0 RBP=000000e164dfa0c0 RSP=000000e164df9b40
 R8=000001eed8ba2280  R9=0000000000000001 R10=0000000000000003 R11=000000e164df9bb0
R12=000000e164df9e70 R13=00007ff78be02b50 R14=000000e164df9f60 R15=000000e164df9f80
-----------------------------------------------------
Stack crawl:
  000000e164df9bc0: 00007ff788b741c0 (not found)
  000000e164df9c00: 00007ff788fccc43 (not found)
  000000e164df9c40: 00007ff78909a636 (not found)
  000000e164df9dc0: 00007ff7890a356a (not found)
  000000e164dff250: 00007ff78963ab96 (not found)
  000000e164dff640: 00007ff7896d4c76 (not found)
  000000e164dff950: 00007ff7896d528a (not found)
  000000e164dff9b0: 00007ff7896352f7 (not found)
  000000e164dffd80: 00007ff78b88ae3e (not found)
  000000e164dffe50: 00007ff7886513c1 (not found)
  000000e164dffe80: 00007ff7886514f6 (not found)
  000000e164dffeb0: 00007fff9c84244d (BaseThreadInitThunk+0x001d)
  000000e164dfff30: 00007fff9ea8dfb8 (RtlUserThreadStart+0x0028)

```

Reply:
Fatal error handler is probably making an assumption about the `running_machine` being set up.  I don’t care too much – anything that throws a fatal error during start should be fixed before it ends up in a release.